t/apicast-policy-rate-limit.t

```diff
@@ -0,0 +1,995 @@
+
+  init_by_lua_block {
+    require "resty.core"
+    ngx.shared.limiter:flush_all()
+    require('apicast.configuration_loader').mock({
+      services = {
+        {
+          id = 42,
+          proxy = {
+            policy_chain = {
+              {
+                name = "apicast.policy.rate_limit",
+                configuration = {
+                  limiters = {
+                    {
+                      name = "fixed_window",
+                      key = "test2",
+                      count = 0,
+                      window = 10
+                    }
+                  },
+                  redis_url = "redis://localhost:6379/1"
+                }
+              }
+            }
+          }
+        }
+      }
+    })
+  }
+  lua_shared_dict limiter 1m;
+
+--- config
+  include $TEST_NGINX_APICAST_CONFIG;
+
+--- request
+GET /
+--- error_code: 500
+--- error_log
+unknown limiter
+
+=== TEST 3: Invalid redis url.
+Return 500 code.
+--- http_config
+  include $TEST_NGINX_UPSTREAM_CONFIG;
+  lua_package_path "$TEST_NGINX_LUA_PATH";
+
+  init_by_lua_block {
+    require "resty.core"
+    ngx.shared.limiter:flush_all()
+    require('apicast.configuration_loader').mock({
+      services = {
+        {
+          id = 42,
+          proxy = {
+            policy_chain = {
+              {
+                name = "apicast.policy.rate_limit",
+                configuration = {
+                  limiters = {
+                    {
+                      name = "connections",
+                      key = "test3",
+                      conn = 20,
+                      burst = 10,
+                      delay = 0.5
+                    }
+                  },
+                  redis_url = "redis://invalidhost:6379/1"
+                }
+              }
+            }
+          }
+        }
+      }
+    })
+  }
+  lua_shared_dict limiter 1m;
+
+--- config
+  include $TEST_NGINX_APICAST_CONFIG;
+
+--- request
+GET /
+--- error_code: 500
+
+=== TEST 4: Rejected (conn) logging only.
+Return 200 code.
+--- http_config
+  include $TEST_NGINX_UPSTREAM_CONFIG;
+  lua_package_path "$TEST_NGINX_LUA_PATH";
+
+  init_by_lua_block {
+    require "resty.core"
+    ngx.shared.limiter:flush_all()
+    require('apicast.configuration_loader').mock({
+      services = {
+        {
+          id = 42,
+          proxy = {
+            policy_chain = {
+              {
+                name = "apicast.policy.rate_limit",
+                configuration = {
+                  limiters = {
+                    {
+                      name = "connections",
+                      key = "test4",
+                      conn = 1,
+                      burst = 0,
+                      delay = 2
+                    }
+                  },
+                  redis_url = "redis://localhost:6379/1",
+                  logging_only = true
+                }
+              },
+              {
+                name = "apicast.policy.rate_limit",
```

Comment:
This policy is duplicated.

Reply:
@davidor I define this policy twice here, because we can open 2 connections by "GET /".